Tamagotchi.Core/Implementations/Dragon/LifecycleService.cs

```diff
@@ -17,51 +16,57 @@ public LifecycleService(IElapsedService elapsedService, ITimeService timeService
+
         public Models.Dragon Hatch(string name, DragonAgeingOptions ageingOptions)
         {
             // Only input here is name
             if (string.IsNullOrWhiteSpace(name))
             {
                 throw new ArgumentException("Parameter name must contain a value", nameof(name));
             }
-            
-            return new()
-            {
-                Name = name,
-                Age = 0,
-                Weight = 10,
-                AgeingOptions = ageingOptions,
-                Happiness = new ActionStatus
-                {
-                    SatisfactionLevel = SatisfactionLevel.Neutral,
-                    LastAction = _timeService.GetCurrentTime(),
-                    LastChecked = _timeService.GetCurrentTime()
-                },
-                Hunger = new ActionStatus
-                {
-                    SatisfactionLevel = SatisfactionLevel.Neutral,
-                    LastAction = _timeService.GetCurrentTime(),
-                    LastChecked = _timeService.GetCurrentTime()
-                },
-                Hatched = _timeService.GetCurrentTime(),
-                LifeStage = DragonLifeStage.Baby
-            };
+
+            return new Models.Dragon(
+                DragonLifeStage.Baby,
+                ageingOptions,
+                name,
+                0,
+                10,
+                 _timeService.GetCurrentTime(),
+                  new ActionStatus
+                  {
+                      SatisfactionLevel = SatisfactionLevel.Neutral,
+                      LastAction = _timeService.GetCurrentTime(),
+                      LastChecked = _timeService.GetCurrentTime()
+                  },
+               new ActionStatus
+               {
+                   SatisfactionLevel = SatisfactionLevel.Neutral,
+                   LastAction = _timeService.GetCurrentTime(),
+                   LastChecked = _timeService.GetCurrentTime()
+               });
         }
 
         public Models.Dragon Age(Models.Dragon tamagotchi)
         {
-            // Update dragons age
-            tamagotchi.Age = (int)_elapsedService.GetElapsedTime(tamagotchi.Hatched).TotalSeconds;
-            
-            // Check status
-            if (tamagotchi.Age > tamagotchi.AgeingOptions.DeadAfter) tamagotchi.LifeStage = DragonLifeStage.Dead;
-            else if (tamagotchi.Age > tamagotchi.AgeingOptions.AdultAfter) tamagotchi.LifeStage = DragonLifeStage.Adult;
-            else if (tamagotchi.Age > tamagotchi.AgeingOptions.TeenAfter) tamagotchi.LifeStage = DragonLifeStage.Teen;
-            else if (tamagotchi.Age > tamagotchi.AgeingOptions.ChildAfter) tamagotchi.LifeStage = DragonLifeStage.Child;
-            else tamagotchi.LifeStage = DragonLifeStage.Baby;
+            return tamagotchi with
+            {
+                Age = (int)_elapsedService.GetElapsedTime(tamagotchi.Hatched).TotalSeconds,
+                LifeStage = Update(tamagotchi.Age, tamagotchi.AgeingOptions)
```

Comment:
@tjmcdonough The use of age here could be problematic - it could create a race condition the tests aren't geared up to capture. Usage of the property before it has been updated with the correct value

Reply:
Yes you are right. Will explain why the tests fail.